dmutils/logging.py

```diff
@@ -125,6 +125,10 @@ def format(self, record):
 
 
 class JSONFormatter(BaseJSONFormatter):
+    def __init__(self, *args, missing_key_attempts=5, **kwargs):
```

Comment:
Might be nice to call this `max_missing_key_attempts` so it reads a bit clearer later on in the code that it's a boundary